File: Leetcode/1765.go
func highestPeak(isWater [][]int) [][]int {
    m:=len(isWater)
    n:=len(isWater[0])
	height := make([][]int, m)
	for i := range height {
		height[i] = make([]int, n)
	}

    cell_queue:=[][]int{} //очередь клеток итераций BFS

    //инициализация карты высот и добавление в очередь клеток с водой
    for i:=0; i<m;i++{
        for j:=0; j<n;j++{
            if isWater[i][j]==0{
                height[i][j]=-1
            }else{
                height[i][j]=0
                cell_queue=append(cell_queue,[]int{i,j})
            }
        }
    }

    for ok := 1; ok != 0; ok = len(cell_queue) {
		x := cell_queue[0][0]
		y := cell_queue[0][1]
		cur_height := height[x][y]
		if (x-1 >= 0 && height[x-1][y] == -1) { //рассматривается клетка сверху
			height[x-1][y] = cur_height + 1
			cell_queue = append(cell_queue, []int{x - 1, y})
		}
		if (x+1 < m && height[x+1][y] == -1) { //рассматривается клетка снизу
			height[x+1][y] = cur_height + 1
			cell_queue = append(cell_queue, []int{x + 1, y})
		}
		if (y-1 >= 0 && height[x][y-1] == -1) { //рассматривается клетка слева
			height[x][y-1] = cur_height + 1
			cell_queue = append(cell_queue, []int{x, y - 1})
		}
		if (y+1 < n && height[x][y+1] == -1) { //рассматривается клетка справа
			height[x][y+1] = cur_height + 1
			cell_queue = append(cell_queue, []int{x, y + 1})
		}
		cell_queue = cell_queue[1:] //убирается из очереди первый элемент
	}
	return height
}

File: Воробьёв/1765.py
from collections import deque
from typing import List


class Solution:
    def highestPeak(self, isWater: List[List[int]]) -> List[List[int]]:
        array = deque([])
        visited = set()
        kor_1, kor_2 = len(isWater), len(isWater[0])
        for i in range(kor_1):
            for j in range(kor_2):
                if isWater[i][j] == 1:
                    array.append((i, j, 0))
                    visited.add((i, j))
        result = [[0 for c in range(kor_2)] for c in range(kor_1)]
        while array:
            size = len(array)
            for c in range(size):
                i, j, height = array.popleft() # удаляет и возвращает первый элемент очереди
                for a, b in [(i + 1, j), (i, j + 1), (i - 1, j), (i, j - 1)]:
                    if 0 <= a < kor_1 and 0 <= b < kor_2 and (a, b) not in visited:
                        visited.add((a, b))
                        array.append((a, b, height + 1))
                        result[a][b] = height + 1
        return result
